python/cuvs/pyproject.toml

```diff
@@ -38,6 +38,7 @@ dependencies = [
     "nvidia-cusolver",
     "nvidia-cusparse",
     "pylibraft==24.10.*,>=0.0.0a0",
+    "rmm==24.10.*,>=0.0.0a0",
```

Comment:
```suggestion
```

I'm not sure about this... it seems to me that `cuvs` doesn't have a direct runtime dependency on the `rmm` Python package.

```shell
git grep -i rmm python/
```

If this wasn't intentional, it should be removed.

Reply:
Agreed. I'll clean this up.

Reply:
I ended up doing a partial cleanup in this PR and a partial cleanup in #322, to avoid merge conflicts.

Reply:
it looks great, thank you for that